### PR TITLE
Update NavigationAgent doc for avoidance rework

### PR DESCRIPTION
**Pull request by smix8**
Reworks NavigationAgent doc for changes in avoidance rework pr https://github.com/godotengine/godot/pull/69988.

_Docs team: this is 4.1+_

### CODE REVIEW

File: tutorials/navigation/navigation_using_navigationagents.rst [hunk 1]
```diff
@@ -3,145 +3,177 @@
 Using NavigationAgents
 ======================
 
-NavigationsAgents are helper nodes to facilitate common calls to the NavigationServer API
-on behalf of the parent actor node in a more convenient manner for beginners.
+NavigationsAgents are helper nodes that combine functionality 
+for pathfinding, pathfollowing and agent avoidance for a Node2D/3D inheriting parent node.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_using_navigationagents.rst:7]:
```suggestion
for pathfinding, path following and agent avoidance for a Node2D/3D inheriting parent node.
```
It irks me that pathfinding is common, but pathfollowing seems to be path following everywhere. 🤷 Anyway, if there are other mentions, those should be changed as well.

File: tutorials/navigation/navigation_using_navigationagents.rst [hunk 2]
```diff
@@ -3,145 +3,177 @@
...
 
 In order for NavigationAgents to use the avoidance feature the ``enable_avoidance`` property must be set to ``true``.
 
 .. image:: img/agent_avoidance_enabled.png
 
+The velocity_computed signal of the NavigationAgent node must be connected to receive the ``safe_velocity`` calculation result.
+
+.. image:: img/agent_safevelocity_signal.png
+
+In order to trigger the avoidance velocity calculation the current velocity of the agent's parent node must be set with ``set_velocity()`` on the NavigationAgent node in ``_physics_process()``.
```
Comment (mhilbrunner) [inline tutorials/navigation/navigation_using_navigationagents.rst:105]:
```suggestion
In order to trigger the avoidance velocity calculation, the current velocity of the agent's parent node must be set with ``set_velocity()`` on the NavigationAgent node in ``_physics_process()``.
```

**Review by mhilbrunner — APPROVED** [review]

**Comment by mhilbrunner** [comment]
Thank you! 🎉

File: tutorials/navigation/navigation_using_navigationagents.rst [hunk 3]
```diff
@@ -3,145 +3,176 @@
...
+New NavigationAgent nodes will automatically join the default navigation map on the World2D/World3D.
 
-.. warning::
-
-    NavigationsAgent nodes and NavigationServer ``agents`` are not the same.
-    The later is an RVO avoidance agent and solely used for avoidance.
-    RVO avoidance agents are not involved in regular pathfinding.
+The functionality of NavigationsAgent nodes can be recreated with scripts and direct
+calls to the NavigationServer API should the default NavigationsAgent 
+nodes not do what is required for a project and specific gameplay.
```
Comment (yedpodtrzitko) [inline tutorials/navigation/navigation_using_navigationagents.rst:19]: @smix8 sorry for tagging you on something what's already merged. Is this sentence phrased as really intended? I dont understand what does it mean circa from the word "should" on second line...

Comment (smix8) [reply tutorials/navigation/navigation_using_navigationagents.rst:19]: It means to role your own agent implementation if the default does not what you need as the NavigationAgent node is optional to work with the navigation system. That block was a copy&paste from the existing doc with not much thought behind. Could be reworded, there are also smaller typos to fix.

**Pull request merged**